### DISCUSSION TITLE
OpenPose Bone Overlay

### DISCUSSION THREAD

**Discussion by if-ai** [Mikubill/sd-webui-controlnet#666] [Q&A]
I am trying to make Goro from MK. I have tried mixing different layers, two openpose CN's, the same layer mixing with canny map I only get one set of arms?

**Comment by if-ai** [comment]
I made this system that allows you to draw extra limbs for your character in Blender but I can't use it 
https://i.imgur.com/nrF0fCh.jpeg

**Reply by ostap667inbox** [reply]
> I made this system that allows you to draw extra limbs for your character in Blender but I can't use it https://i.imgur.com/nrF0fCh.jpeg

This is NOT a ControlNet problem, but a problem of model quality, your prompt and weight of all influencing factors like negative prompts, LoRa, textual invertion, etc. Models, on the contrary, are trained hard, so that they do not give birth to mutants, while you are trying to overcome this :)

But anything is possible. Here is a simple prompt 'four handed monster', the ControlNet weight is increased to 2, and no other quality-enhancing additions.
And here is the result with four hands:

![4hands](https://user-images.githubusercontent.com/123593542/229991440-1d52d7ac-41a4-4cb1-b14c-72e9e2d593c4.jpg)

**Reply by ostap667inbox** [reply]
No problem with 4 hands.
RPG4:
_(((four handed))) warrior, skin pores, (((bare arms))), (((beard))), wlop, ilya kuvshinov, and greg rutkowski, (((white background))), a sling, a belt, a loincloth
Negative prompt: armor, protection, shoulder pads, shield, kneepads, wool, fur, (((props)))_
![4hands-2](https://user-images.githubusercontent.com/123593542/229994995-c184e139-8600-455a-aff9-a2a10079eab4.jpg)

**Reply by ostap667inbox** [reply]
You may have a problem with the color of the joints on your skeleton. For example, in your screenshot, I see differences in the colors of the same shoulder joint for the two left hands. For my examples I used the A1111 extension '3D Openpose'. First, I made a picture with two arms pose. Then for the same pose and with the same camera position I changed only the hands. Then in photoshop the second picture was superimposed on top of the first one with 'Lighten' overlay. So the color of the bones and joints on the skeleton remained correct.

**Reply by ostap667inbox** [reply]
The ControlNet weight = 2 is an important parameter to defeat attempts to replace the 'wrong' number of limbs with other objects or background elements when generating.
![4hands-3](https://user-images.githubusercontent.com/123593542/229999402-5a666540-ce61-4df1-a1f7-a8065cdae1a5.jpg)

**Reply by if-ai** [reply]
> You may have a problem with the color of the joints on your skeleton. For example, in your screenshot, I see differences in the colors of the same shoulder joint for the two left hands. For my examples I used the A1111 extension '3D Openpose'. First, I made a picture with two arms pose. Then for the same pose and with the same camera position I changed only the hands. Then in photoshop the second picture was superimposed on top of the first one with 'Lighten' overlay. So the color of the bones and joints on the skeleton remained correct.

Thank you for your test, and yes you are correct. I mixed the color of the wrist.
Like you said the trick is to set the weight to 2. 
here is the result on my video https://youtu.be/ZRQKAOD4ml0?t=2713, 

I want to use the system of extra bones to make things like a centaur a dragon or alien creatures and I wonder if it can even do that because of the model as well.  

Recently, I received new information about the Openpose bone model and my rig has many issues that I will be fixing before the next update and test the system again and see if that might improve things on the extra limbs too.

Again thank you for all the pointers you gave me here.

**Reply by ostap667inbox** [reply]
I believe that a bone system is not enough for such complex objects. A lot of different fun combinations of several characters will be generated, and dogs and other animals familiar to this model will be added. It makes sense to use Multi-ControlNet and add a layer with scribbles or/and depth map in addition to bones.
Adjusting the Guidance Start and Guidance End parameters can also give very good results. Guidance Start = 0 and Guidance End = 1 affect the generation process from start to finish. This often makes the final result worse. Decreasing Guidance End to 0.5-0.6 will allow you to make changes in the generation process, but the final part of the generation will be done without being affected by additional noise.

**Reply by if-ai** [reply]
That's an interesting approach I will try it with those settings. here is a post by @toyxyz on twitter he did manage to do something closer to what I want to achieve https://twitter.com/toyxyz3/status/1627696689535664128/photo/1 right now and since you mention the depth maps it gave me the idea of perhaps adding animal parts the way I did the weapons and props in my rig it looks like is the only way to achieve satisfactory results would be with actual geometry depth maps.
